### PR TITLE
Bug Fixes for Nested Compositor

### PR DESCRIPTION
**Pull request by JPEWdev**
A few minor bug fixes. Let me know what you think about the change to move the wayland poll() loop onto the threaded compositor thread. This change was required for WebKit to work with mesa (although it work fine on my embedded hardware without it for whatever reason).

### CODE REVIEW

File: Source/WebKit2/Shared/CoordinatedGraphics/threadedcompositor/ThreadedCompositor.cpp [hunk 1]
```diff
@@ -320,7 +320,6 @@ static void debugThreadedCompositorFPS()
 
 void ThreadedCompositor::frameComplete()
 {
-    ASSERT(&RunLoop::current() == &m_compositingRunLoop->runLoop());
```
Comment (zdobersek) [inline Source/WebKit2/Shared/CoordinatedGraphics/threadedcompositor/ThreadedCompositor.cpp]: frameComplete() should be called on the threaded compositor thread.

Was this assert failing or just not compiling?

Comment (JPEWdev) [inline Source/WebKit2/Shared/CoordinatedGraphics/threadedcompositor/ThreadedCompositor.cpp]: It wasn't compiling, runLoop is not a member of m_compositingRunLoop(). It looks like it was reworked previously. I tried to get the SHA-1 where it was change in the commit message of 	3733214, but I failed and got the wrong one.

Comment (JPEWdev) [inline Source/WebKit2/Shared/CoordinatedGraphics/threadedcompositor/ThreadedCompositor.cpp]: a15a9620d67e5638cd67944ac30f0958e45db9b4 appears to be the the correct commit that broke it

Comment (zdobersek) [inline Source/WebKit2/Shared/CoordinatedGraphics/threadedcompositor/ThreadedCompositor.cpp]: Fixed it in https://github.com/WebKitForWayland/webkit/commit/8e168cbc57cc7d606d7cf0cbc156351226c6f646. Feel free to pull from master to get your branch up-to-date.

Comment (JPEWdev) [inline Source/WebKit2/Shared/CoordinatedGraphics/threadedcompositor/ThreadedCompositor.cpp]: Done

**Pull request merged**